modules/core/src/main/java/org/apache/synapse/mediators/v2/ScatterGather.java

```diff
@@ -0,0 +1,821 @@
+        Object o = axis2MessageCtx.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+        Map headers = (Map) o;
+        if (headers != null) {
+            headers.put(HTTP.CONTENT_TYPE, resultValue);
+        }
+    }
+
+    private static void addChildren(List list, OMElement element) {
+
+        for (Object item : list) {
+            if (item instanceof OMElement) {
+                element.addChild((OMElement) item);
+            }
+        }
+    }
+
+    private static List getMatchingElements(MessageContext messageContext, SynapsePath expression) {
+
+        Object o = expression.objectValueOf(messageContext);
+        if (o instanceof OMNode) {
+            List list = new ArrayList();
+            list.add(o);
+            return list;
+        } else if (o instanceof List) {
+            return (List) o;
+        } else {
+            return new ArrayList();
+        }
+    }
+
+    private void setAggregatedMessageAsVariable(MessageContext originalMessageContext, Aggregate aggregate) {
+
+        Object variable = null;
+        if (Objects.equals(contentType, JSON_TYPE)) {
+            log.debug("Merging aggregated JSON responses to variable");
+            variable = new JsonArray();
+            setJSONResultToVariable((JsonArray) variable, aggregate);
+        } else if (Objects.equals(contentType, XML_TYPE)) {
+            log.debug("Merging aggregated XML responses to variable");
+            variable = OMAbstractFactory.getOMFactory().createOMElement(new QName(rootElementName));
+            setXMLResultToRootOMElement((OMElement) variable, aggregate);
+        } else {
+            handleException(aggregate, "Error merging aggregation results to variable : " + resultTarget +
+                    " unknown content type : " + contentType, null, originalMessageContext);
+        }
+        originalMessageContext.setVariable(resultTarget, variable);
+        StatisticDataCollectionHelper.collectAggregatedParents(aggregate.getMessages(), originalMessageContext);
+    }
+
+    private void setJSONResultToVariable(JsonArray variable, Aggregate aggregate) {
+
+        for (MessageContext synCtx : aggregate.getMessages()) {
+            Object evaluatedResult = aggregationExpression.objectValueOf(synCtx);
+            variable.add((JsonElement) evaluatedResult);
+        }
+    }
+
+    private void setXMLResultToRootOMElement(OMElement element, Aggregate aggregate) {
+
+        for (MessageContext synCtx : aggregate.getMessages()) {
+            List list = getMatchingElements(synCtx, aggregationExpression);
+            addChildren(list, element);
+        }
+    }
+
+    private MessageContext getAggregatedMessage(Aggregate aggregate) {
+
+        MessageContext newCtx = null;
+        if (Objects.equals(contentType, JSON_TYPE)) {
+            JsonArray jsonArray = new JsonArray();
+                log.debug("Merging aggregated JSON responses to body");
```

Comment:
should we check before log debug

Reply:
Since we are not doing any string concatenations, we don't need to check debugEnabled